api/s2n.h

```diff
@@ -1871,6 +1872,21 @@ S2N_API extern int s2n_connection_prefer_low_latency(struct s2n_connection *conn
  */
 S2N_API extern int s2n_connection_set_recv_buffering(struct s2n_connection *conn, bool enabled);
 
+/**
+ * Reports how many bytes of encrypted data is buffered due to the optimization
+ * enabled by `s2n_connection_set_recv_buffering`.
+ *
+ * While `s2n_peek` reports decrypted data that is ready for the application to consume,
+ * `s2n_peek_buffered` reports encrypted data that needs to be parsed and decrypted before
```

Comment:
It might be better to talk about TLS record/message framing than encrypted/decrypted status. Like there could be only transport-level messages pending that results in no actual application bytes being read.